### ISSUE TITLE
no-string-based-set-timeout on use Socket.setTimeout

### ISSUE THREAD

**Issue by SanderDeWaal1992** [microsoft/tslint-microsoft-contrib#442]
This piece of code does result in a `no-string-based-set-timeout` lint error:
```
const client = new net.Socket();
client.setTimeout(2000, () => {
	// nothing
});
```
The lint error I get:
`ERROR: (no-string-based-set-timeout) C:/...: Forbidden setTimeout string parameter: 2000`

Using typescript 9.4.

Am I correct that this is a false positive?

**Comment by JoshuaKGoldberg** [comment]
Indeed, looks like a false positive. A few ways to make this better...
* The rule probably shouldn't complain for non-strings, such as numbers.
* The rule could become an optionally [typed rule](https://palantir.github.io/tslint/usage/type-checking/) and, when given a TS program, could use that to verify whether the method actually has a method signature similar to `setTimeout`

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #876. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!
